codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryAttribute.java

```diff
@@ -122,27 +122,29 @@ public String toString() {
 
     @Override
     public Attribute copy() {
-        MemoryAttribute attr = new MemoryAttribute(getName());
-        attr.setCharset(getCharset());
-        ByteBuf content = content();
-        if (content != null) {
-            try {
-                attr.setContent(content.copy());
-            } catch (IOException e) {
-                throw new ChannelException(e);
-            }
-        }
-        return attr;
+        final ByteBuf content = content();
+        return replace(content != null ? content.copy() : null);
     }
 
     @Override
     public Attribute duplicate() {
+        final ByteBuf content = content();
+        return replace(content != null ? content.duplicate() : null);
+    }
+
+    @Override
+    public Attribute retainedDuplicate() {
+        final ByteBuf content = content();
+        return replace(content != null ? content.retainedDuplicate() : null);
```

Comment:
if a `ChannelException` is thrown we need to take care to release the `retained` buffer in this case